datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -216,6 +218,8 @@ impl CoalesceBatchesStream {
             match input_batch {
                 Poll::Ready(x) => match x {
                     Some(Ok(batch)) => {
+                        let batch = gc_string_view_batch(&batch);
```

Comment:
So here inside `gc` string buffer will be copied once, (below) in `concat_batches()` string buffer will be copied again, it seems possible to copy only once by changing the internal implementation of `concat_batches()`
But I think we can do it later when there is a good benchmark to assess the impact, at least excessive copy in `coalesce_batches()` does not have a huge performance impact on TPCH now.

Reply:
This is an excellent point

I think given how concat is implemented for `StringView` it will only copy the fixed parts (not the actual string data)

Perhaps what we could do is implement a wrapper around arrow::concat_batches that has the datafusion specific GC trigger for sparse arrays, and falls back to concat for other types: https://docs.rs/arrow-select/52.1.0/src/arrow_select/concat.rs.html#150

```rust
/// wrapper around [`arrow::compute::concat`] that 
pub fn concat(arrays: &[&dyn Array]) -> Result<ArrayRef, ArrowError> {
 // loop over columns here and handle StringView specially, 
 // or fallback to concat
 }
```

Reply:
I filed https://github.com/apache/datafusion/issues/11628 to track this idea